tests/scripts/run_tests.sh

```diff
@@ -103,85 +105,102 @@ AddHeader "X-My-Header3" "Powered by Tinyproxy"
 Upstream http 255.255.255.255:65535 ".invalid"
 EOF
 
-cat << 'EOF' > $TINYPROXY_FILTER_FILE
-.*\.google-analytics\.com$
+cat << 'EOF' > "$TINYPROXY_FILTER_FILE"
+.*\.google-analytics\.com\$
 EOF
 }
 
 start_tinyproxy() {
-	echo -n "starting tinyproxy..."
-	$VALGRIND $TINYPROXY_BIN -c $TINYPROXY_CONF_FILE 2> $TINYPROXY_STDERR_LOG
-	echo " done (listening on $TINYPROXY_IP:$TINYPROXY_PORT)"
+	printf "starting tinyproxy..."
+	$VALGRIND "$TINYPROXY_BIN" -c "$TINYPROXY_CONF_FILE" 2> "$TINYPROXY_STDERR_LOG"
+	echo " done. (listening on $TINYPROXY_IP:$TINYPROXY_PORT)"
 }
 
 reload_config() {
-	echo -n "signaling tinyproxy to reload config..."
-	pid=$(cat $TINYPROXY_PID_FILE)
+	printf "signaling tinyproxy to reload config..."
+    local pid 
+	pid=$(cat "$TINYPROXY_PID_FILE")
 	#1: SIGHUP
-	kill -1 $pid && echo "ok" || echo "fail"
+	kill -1 "$pid" && echo "ok" || echo "fail"
 }
 
 stop_tinyproxy() {
-	echo -n "killing tinyproxy..."
-	pid=$(cat $TINYPROXY_PID_FILE)
-	kill $pid
-	if test "x$?" = "x0" ; then
+	printf "killing tinyproxy..."
+    local pid 
```

Comment:
fwiw, even dash supports local

Reply:
yeah, I wasn't removing local. just changing the way it is used. shellcheck said to separate declaration from assignment.